sstables/sstables_manager_subscription.hh

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#pragma once
+
+#include <boost/signals2/dummy_mutex.hpp>
+#include <boost/signals2/signal_type.hpp>
+
+#include "sstables/generation_type.hh"
+
+namespace bs2 = boost::signals2;
+
+namespace sstables {
+
+enum class manager_event_type {
+    add,
+    unlink
+};
+using manager_signal_callback_type = std::function<future<>(sstables::generation_type, manager_event_type)>;
```

Comment:
Based on clients real needs we may want in the future to separate into a callable-per-event instead of carrying the event-type within a single callable.
For example if the context of the registrar is different for `added` and `unlinked` we might prefer having two separate calls.

Reply:
The original version of this subscription was structured like that.
It is a bit more complicated, but doable.

Reply:
defined structured subscription class in v5 ([a8bc344](https://github.com/scylladb/scylladb/commit/a8bc344156a32b48877ac1fa90736f07f9d77a6c))
using separate methods for added vs. deleted sstable notification